src/Structures/CollectionTree.php

```diff
@@ -21,16 +21,22 @@ public static function fromModel(Model $model)
     }
 
     public function toModel()
+    {
+        return self::makeModelFromContract($this);
+    }
+
+    public static function makeModelFromContract($source)
     {
         $class = app('statamic.eloquent.collections.tree_model');
 
-        return $class::findOrNew($this->model?->id)->fill([
-            'handle' => $this->handle(),
-            'initial_path' => $this->initialPath(),
-            'locale' => $this->locale(),
-            'tree' => $this->tree(),
-            'type' => 'collection',
-        ]);
+        return $class::findOrNew($source instanceof \Statamic\Structures\CollectionTree ? null : $this->model?->id)
```

Comment:
@ryanmitchell I think this should also be imported `\Statamic\Structures\CollectionTree` to do it everywhere the same way? (Only a style thing).

Reply:
Thanks, updated